beacon_node/beacon_chain/src/block_verification.rs

```diff
@@ -1302,6 +1308,94 @@ impl<T: BeaconChainTypes> ExecutionPendingBlock<T> {
 
         check_block_relevancy(block.as_block(), block_root, chain)?;
 
+        // Spawn an async call to the execution node to fetch blobs.
+        let blob_chain = chain.clone();
+        let blob_block = block.as_block().clone();
+        let blob_fetcher_future = async move {
+            let block = blob_block;
+            let chain = blob_chain;
+            let execution_payload = block.message().execution_payload()?;
+            let blob_ids = extract_blob_transaction_ids::<T::EthSpec>(
+                execution_payload.transactions().unwrap(),
+            )
+            .map_err(|e| {
+                BlockError::ExecutionPayloadError(ExecutionPayloadError::RequestFailed(
+                    execution_layer::Error::VerifyingVersionedHashes(e),
+                ))
+            })?;
+            let execution_layer = chain
+                .execution_layer
+                .as_ref()
+                .ok_or(BeaconChainError::ExecutionLayerMissing)?;
+            debug!(
+                chain.log,
+                "Blobs from EL request";
+                "num_blob_tx" => blob_ids.len(),
+            );
+            let response = execution_layer.get_blobs(blob_ids).await.map_err(|e| {
+                BlockError::ExecutionPayloadError(ExecutionPayloadError::RequestFailed(e))
+            })?;
+            let num_success = response.blobs.iter().filter(|b| b.is_some()).count();
+            if num_success == 0 {
+                debug!(chain.log, "Blobs from EL empty");
+            }
+            debug!(
+                chain.log,
+                "Blobs from EL success";
+                "num_blob_tx" => num_success
+            );
+            let mut fixed_blob_sidecar_list = FixedBlobSidecarList::default();
+            for (i, (blob, kzg_proof)) in response
+                .blobs
+                .into_iter()
+                .flatten()
+                .flat_map(|blob| blob.blobs.into_iter().zip(blob.proofs))
+                .enumerate()
+            {
+                match BlobSidecar::new(i, blob, &block, kzg_proof) {
```

Comment:
TODO: should use KZG inclusion proof from EL rather than computing it here

need to add verification to make sure it is valid

Reply:
nvm, the EL doesn't seem to know the KZG inclusion proof